xarray_filters/datasets.py

```diff
@@ -1,117 +1,415 @@
+    --------
+    {skl_funcname}
+    {xy_transformer}
+
+    """.format(
+            skl_funcname=(skl_sampler_func.__module__ + '.' + skl_sampler_func.__name__),
+            xy_transformer=(XyTransformer.__module__ + '.' + XyTransformer.__name__),
+            accepted_types=(str(XyTransformer.accepted_types))
+
+    )
+    wrapper.__doc__ = preamble_doc # + skl_sampler_func.__doc__
+    # Task 2 of 2: fixing the signature of `wrapper`
+    astype_param = inspect.Parameter(name='astype', kind=inspect.Parameter.KEYWORD_ONLY, default=default_astype)
+    kwargs_param = inspect.Parameter(name='kwargs', kind=inspect.Parameter.VAR_KEYWORD)
+    params = skl_params + [astype_param, kwargs_param]
+    wrapper.__signature__ = inspect.Signature(params)
+    wrapper.__name__ = skl_sampler_func.__name__
+    return wrapper
+
+
+def fetch_lfw_people(*args, **kw):
+    '''TODO Gui wrap docs from sklearn equivalent
+    and add notes about it returning MLDataset
+
+    out = fetch_lfw_people()
+    In [6]: out.ariel_sharon_65
+Out[6]:
+<xarray.DataArray 'ariel_sharon_65' (row: 62, column: 47)>
+array([[ 111.666664,  128.333328,  123.333336, ...,  186.333328,  188.      ,
+         188.333328],
+       [ 103.      ,  124.666664,  121.      , ...,  184.333328,  186.      ,
+         187.333328],
+       [  97.      ,  119.333336,  114.666664, ...,  178.333328,  180.333328,
+         183.333328],
+       ...,
+       [  23.666666,   24.      ,   20.      , ...,  182.      ,  192.666672,
+         200.      ],
+       [  21.333334,   20.666666,   18.      , ...,  191.      ,  201.      ,
+         202.666672],
+       [  20.666666,   18.      ,   14.666667, ...,  197.333328,  202.      ,
+         199.333328]], dtype=float32)
+Coordinates:
+  * row      (row) int64 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 ...
+  * column   (column) int64 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 ...
+Attributes:
+    id:       373
+    name:     Ariel Sharon
+    '''
+    out = sklearn.datasets.fetch_lfw_people(*args, **kw)
+    dset = OrderedDict()
+    name_tracker = defaultdict(lambda: -1)
+    for img, name_id in zip(out.images, out.target):
+        name = out.target_names[name_id]
+        name_tracker[name] += 1
+        clean_name = name.lower().replace(' ', '_') + '_' + str(name_tracker[name])
+        r, c = img.shape
+        coords = OrderedDict([('row', np.arange(r)),
+                              ('column', np.arange(c))])
+        dims = ('row', 'column',)
+        attrs = dict(name=name, id=name_id)
+        data_arr = xr.DataArray(img, coords=coords, dims=dims, attrs=attrs)
+        dset[clean_name] = data_arr
+    attrs = dict(metadata=[args, kw])
+    # TODO - PR 3's MLDataset instead of Dataset
+    dset = xr.Dataset(dset, attrs=attrs)
+    # TODO - allow converting to dataframe, numpy array?
+    return dset
+
+
+# Convert all sklearn functions that admit conversion
+converted_make_funcs = dict()  # holds converted sklearn funcs
```

Comment:
`converted_make_funcs` and `sklearn_make_funcs` should probably be "hidden" with leading underscores.